### PR TITLE
convert pointer to ptr for job controller

### PR DESCRIPTION
**Pull request by kannon92**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
pointer is deprecated so I went ahead and converted to use ptr in the job controller.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/job/job_controller_test.go [hunk 1]
```diff
@@ -60,6 +60,7 @@ import (
 	"k8s.io/utils/clock"
 	clocktesting "k8s.io/utils/clock/testing"
 	"k8s.io/utils/pointer"
```
Comment (mimowo) [inline pkg/controller/job/job_controller_test.go:62]: Please remove all uses so that this import can be dropped

**Review by alculquicondor — COMMENTED** [review]
/approve

**Pull request merged**